### PR TITLE
Reserve URLs using Publishing API...

### PR DESCRIPTION
**Pull request by steventux**
Another part of this story https://trello.com/c/bf4nuPIc/340-move-url-arbitration-to-publishing-api
Path reservation is no longer the responsibility of URL Arbiter,
this change makes Panopticon reserve paths using the Publishing API endpoint.

### CODE REVIEW

File: test/integration/artefacts_api_test.rb [hunk 1]
```diff
@@ -268,6 +268,7 @@ class ArtefactsAPITest < ActiveSupport::TestCase
         put "/artefacts/wibble.json", artefact_data
 
         assert_equal 409, last_response.status
+        puts last_response.body.strip
```
Comment (elliotcm) [inline test/integration/artefacts_api_test.rb]: You probably don't want to leave this.

**Comment by elliotcm** [comment]
All :+1: except one debugging statement left in.

File: lib/tasks/detailed_guides.rake [hunk 2]
```diff
@@ -10,21 +10,21 @@ namespace :detailed_guide do
...
           detailed_guide.set(:slug, new_slug)
           puts "Renamed #{old_slug} to #{new_slug} (#{i+1}/#{count})"
-        rescue GOVUK::Client::Errors::Conflict => e
+        rescue GdsApi::HTTPClientError => e
           puts "Couldn't rename #{old_slug} to #{new_slug} (#{i+1}/#{count})"
           message = ""
-          if e.response["errors"]
-            e.response["errors"].each do |field, errors|
+          if e.error_details["errors"]
+            e.error_details["errors"].each do |field, errors|
```
Comment (benlovell) [inline lib/tasks/detailed_guides.rake]: You could golf this down a little and remove the conditional by normalising 

```ruby
Array(e.error_details["errors"]).each #...
```

But it's purely a matter of opinion.

**Comment by steventux** [comment]
@elliotcm thanks for catching that, updated.

**Pull request merged**